src/vs/workbench/contrib/terminalContrib/stickyScroll/browser/terminalStickyScrollOverlay.ts

```diff
@@ -193,6 +194,19 @@ export class TerminalStickyScrollOverlay extends Disposable {
 	}
 
 	private _refresh(): void {
+		this._refreshTimeout.clear();
+
+		// Always debounce by 100ms to prevent flashing during rapid scroll/enter events
```

Comment:
I've tried wrapping this timeout conditionally for only pager related scenarios, using https://github.com/microsoft/vscode/blob/a3208e218c788bc19d9c915c4612d37ae2650096/src/vs/workbench/contrib/terminalContrib/stickyScroll/browser/terminalStickyScrollOverlay.ts#L280 , but found out the current code (Pre-PR) doesn't always detect we are in pager as I enter through the `git diff`.

Reply:
`_refresh` actually already has some debouncing but it's at the microtask level:

https://github.com/microsoft/vscode/blob/fb3ad8dfeb2118c3dbf5154c89be35cdf8cb9e58/src/vs/workbench/contrib/terminalContrib/stickyScroll/browser/terminalStickyScrollOverlay.ts#L195-L204

You want to debounce the hide -> show transition specifically. And for that you should use the `@debounce` helper